pkg/storage/cacher.go

```diff
@@ -150,14 +153,31 @@ func NewCacherFromConfig(config CacherConfig) *Cacher {
 		watchers:   make(map[int]*cacheWatcher),
 		versioner:  config.Versioner,
 		keyFunc:    config.KeyFunc,
+		stopped:    false,
+		// We make stopCh a buffered channel, because we potentially need to stop both:
+		// - util.Until go-routine
+		// - reflector.ListAndWatch
+		// but there are no guarantees to the order that we need to stop them.
```

Comment:
...there are no guarantees on the order that they will stop, so we make the channel...